Core/Core/Inbox/MessageDetails/ViewModel/MessageDetailsViewModel.swift

```diff
@@ -0,0 +1,121 @@
+    private let interactor: MessageDetailsInteractor
+    private let router: Router
+    private let myID: String
+
+    public init(router: Router, interactor: MessageDetailsInteractor, myID: String) {
+        self.interactor = interactor
+        self.router = router
+        self.myID = myID
+
+        setupOutputBindings()
+        setupInputBindings(router: router)
+    }
+
+    public func moreTapped(viewController: WeakViewController) {
+        let sheet = BottomSheetPickerViewController.create()
+        sheet.addAction(
+            image: .replyLine,
+            title: NSLocalizedString("Reply", bundle: .core, comment: ""),
+            accessibilityIdentifier: "MessageDetails.reply"
+        ) {}
+        sheet.addAction(
+            image: .replyAllLine,
+            title: NSLocalizedString("Reply All", bundle: .core, comment: ""),
+            accessibilityIdentifier: "MessageDetails.replyAll"
+        ) {}
+
+        sheet.addAction(
+            image: .forwardLine,
+            title: NSLocalizedString("Forward", bundle: .core, comment: ""),
+            accessibilityIdentifier: "MessageDetails.markAllRead"
+        ) {}
+
+        sheet.addAction(
+            image: .archiveLine,
+            title: NSLocalizedString("Archive", bundle: .core, comment: ""),
+            accessibilityIdentifier: "MessageDetails.archive"
+        ) {}
+
+        sheet.addAction(
+            image: .trashLine,
+            title: NSLocalizedString("Delete", bundle: .core, comment: ""),
+            accessibilityIdentifier: "MessageDetails.delete"
+        ) {}
+        router.show(sheet, from: viewController, options: .modal())
+    }
+
+    public func replyTapped(viewController: WeakViewController) {}
+
+    private func setupOutputBindings() {
+        interactor.state
+                .assign(to: &$state)
+        interactor.subject
+            .map { subject in
+                subject.isEmpty ? NSLocalizedString("No Subject", comment: "") : subject
+            }
+            .assign(to: &$subject)
+        interactor.messages
+            .map { messages in
+                messages.map {
+                    MessageViewModel(item: $0, myID: self.myID, userMap: self.interactor.userMap)
+                }
+            }
+            .assign(to: &$messages)
+        interactor.starred
+            .assign(to: &$starred)
+    }
+
+    private func setupInputBindings(router: Router) {
+        let interactor = self.interactor
+        refreshDidTrigger
+            .delay(for: .seconds(1), scheduler: RunLoop.main)
+            .flatMap { refreshCompletion in
+                interactor
+                    .refresh()
+                    .receive(on: DispatchQueue.main)
+                    .handleEvents(receiveOutput: { refreshCompletion() })
+            }
+            .sink()
+            .store(in: &subscriptions)
+        starDidTap
+            .map { _ in
+                interactor.updateStarred(starred: !self.starred) }
```

Comment:
I think this is a retain cycle. We store the stream in `subscriptions` and the stream strong references `self`. I think we could safely capture self unowned here. (When the viewmodel is deallocated the user can't tap the star icon.)